backend/src/crd/controller/scheduledworkflow/controller.go

```diff
@@ -439,11 +439,14 @@ func (c *Controller) syncHandler(key string) (
 // an error (if any), and a boolean indicating (in case of an error) whether handling the
 // ScheduledWorkflow should be attempted again at a later time.
 func (c *Controller) submitNextWorkflowIfNeeded(swf *util.ScheduledWorkflow,
-		activeWorkflowCount int, nowEpoch int64) (
-		workflow *commonutil.Workflow, nextScheduledEpoch int64, err error) {
+	activeWorkflowCount int, nowEpoch int64) (
+	workflow *commonutil.Workflow, nextScheduledEpoch int64, err error) {
 	// Compute the next scheduled time.
-	nextScheduledEpoch, shouldRunNow := swf.GetNextScheduledEpoch(
+	nextScheduledEpoch, shouldRunNow, err := swf.GetNextScheduledEpoch(
```

Comment:
Can `GetNextScheduledEpoch` return an error?
I believe in previous rounds of changes, it should no longer emit error. Did I recall correctly?

If so, I think we can also remove most of the changes related to adding the error argument.

Reply:
> Right now, I worry this change is too big thus may more likely contain bugs our tests did not cover, so I'm trying pushing for removing all the unnecessary changes out of this PR to minimize the risk.

Sounds good, also always appreciate the feedback, so no worries!  

When we moved the location to the controller, we removed the error handling since we did not need to pass it up. Now when I changed it to the CronSchedule the error that can come from loading the location should be propagate upwards, I might have misunderstood your suggestion though?

Since we now set the location inside the `NewCronSchedule` and we return an error during 
```
	location, err := GetLocation()
```
this will result in that we have: 
```
schedule, err := NewCronSchedule(s.Spec.Trigger.CronSchedule)
```
and thus we need to handled this in 

```
nextScheduledEpoch, shouldRunNow, err := swf.GetNextScheduledEpoch(
```

As I see it we either need to pass the location down or the error up in this case.

Reply:
Good point! I totally missed that. I'll take another look tomorrow

Reply:
Hi @NikeNano, sorry for the delay. I just reviewed the new structure, I believe the previous reasons for moving location initialization to program start are still valid.
Even though we decided to put location as a field for cronschedule, we can initialize location at the start of the controller and pass location as an argument to `NewCronSchedule` constructor, this way, we can still avoid handling so many errors throughout the code base.

Reply:
> we move it back to the controller and initialise it once, when the controller starts. I believe this solution is actually better.

@NikeNano Agree with you.

My apologies, my suggestion in https://github.com/kubeflow/pipelines/pull/4641#discussion_r530378566 was wrong, because I didn't realize `NewCronSchedule` was called inside `GetNextScheduledEpoch`.

Reply:
No worries :) , will revert back.

Reply:
Could you please take a look again @Bobgy?